oops/meshgrid.py

```diff
@@ -88,48 +109,137 @@ def for_fov(fov, origin=0.5, undersample=1, oversample=1, limit=None,
                         the properties of the FOV.
         """
 
+        u_axis, v_axis = (1,0) if swap else (0,1)
+        return Meshgrid.for_shape(fov, fov.uv_shape.vals, u_axis, v_axis,
+                                  origin=origin,
+                                  undersample=undersample,
+                                  oversample=oversample,
+                                  limit=limit,
+                                  fov_keywords=fov_keywords)
+
+    #===========================================================================
+    @staticmethod
+    def for_shape(fov, shape, u_axis=-1, v_axis=-1, origin=None, undersample=1,
+                  oversample=1, limit=None, center_uv=None, fov_keywords={}):
+        """A 2-D rectangular Meshgrid object for a specified FOV and uv_shape.
+
+        Input:
+            fov         FOV object.
+
+            shape       overall shape to which this Meshgrid must broadcast.
+
+            u_axis      location of the u axis within the shape; -1 if there is
+                        no u-axis.
+
+            v_axis      location of the v axis within the shape; -1 if there is
+                        no v-axis.
+
+            origin      A single value, tuple or Pair defining the (u,v) origin
+                        of the grid. Default is to place the first sample in the
+                        middle of the first pixel (after allowing for the under-
+                        or oversampling).
+
+            undersample A single value, tuple or Pair defining the magnitude of
+                        under-sampling to be performed. For example, a value of
+                        2 would cause the meshgrid to sample every other pixel
+                        along each axis.
+
+            oversample  A single value, tuple or Pair defining the magnitude of
+                        over-sampling to be performed. For example, a value of
+                        2 would create a 2x2 array of samples inside each pixel.
+
+            limit       A single value, tuple or Pair defining the (u,v) upper
+                        limits of the meshgrid. By default, this is the shape of
+                        the FOV.
+
+            center_uv   A single value, tuple or Pair defining the (u,v) center
+                        of the FOV. Default is to place this point at the center
+                        of the specified grid of points.
+
+            fov_keywords  an optional dictionary of parameters passed to the
+                        FOV methods, containing parameters that might affect
+                        the properties of the FOV.
+        """
+
+        if undersample < 1:
```

Comment:
These checks crash. undersample and oversample can be tuples or Pairs, which can't be compared with `< 1`.

Reply:
Fixed.